assessment/collectors/infoschema_collector_test.go

```diff
@@ -0,0 +1,713 @@
+	for _, trigger := range triggers {
+		assert.Contains(t, []string{"trigger1", "trigger2"}, trigger.Name)
+		assert.Equal(t, "table1", trigger.TargetTable)
+		assert.Equal(t, "table1", trigger.TargetTableId)
+	}
+}
+
+func TestInfoSchemaCollector_ListFunctions(t *testing.T) {
+	mockConv := &internal.Conv{}
+	collector := InfoSchemaCollector{
+		conv: mockConv,
+		functions: []utils.FunctionAssessmentInfo{
+			{Name: "function1", Definition: "CREATE FUNCTION function1() RETURNS INT RETURN 1;"},
+			{Name: "function2", Definition: "CREATE FUNCTION function2() RETURNS INT RETURN 2;"},
+		},
+	}
+	functions := collector.ListFunctions()
+	assert.Len(t, functions, 2)
+	for _, function := range functions {
+		assert.Contains(t, []string{"function1", "function2"}, function.Name)
+		assert.Equal(t, 1, function.LinesOfCode)
+	}
+}
+
+func TestInfoSchemaCollector_ListViews(t *testing.T) {
+	mockConv := &internal.Conv{
+		UsedNames: make(map[string]bool),
+	}
+	collector := InfoSchemaCollector{
+		conv: mockConv,
+		views: []utils.ViewAssessmentInfo{
+			{Name: "view1", Definition: "SELECT * FROM table1"},
+			{Name: "view2", Definition: "SELECT col1, col2 FROM table2 WHERE col3 > 10"},
+		},
+	}
+	views := collector.ListViews()
+	assert.Len(t, views, 2)
+	for _, view := range views {
+		assert.Contains(t, []string{"view1", "view2"}, view.SrcName)
+		assert.Equal(t, "NON-MATERIALIZED", view.SrcViewType)
+		assert.NotEmpty(t, view.SpName)
+	}
+}
+
+func TestInfoSchemaCollector_ListStoredProcedures(t *testing.T) {
+	mockConv := &internal.Conv{}
+	collector := InfoSchemaCollector{
+		conv: mockConv,
+		storedProcedures: []utils.StoredProcedureAssessmentInfo{
+			{Name: "procedure1", Definition: "CREATE PROCEDURE procedure1() BEGIN SELECT 1; END;"},
+			{Name: "procedure2", Definition: "CREATE PROCEDURE procedure2() BEGIN SELECT 2; END;"},
+		},
+	}
+	sps := collector.ListStoredProcedures()
+	assert.Len(t, sps, 2)
+	for _, sp := range sps {
+		assert.Contains(t, []string{"procedure1", "procedure2"}, sp.Name)
+		assert.Equal(t, 2, sp.LinesOfCode)
+	}
+}
+
+func TestInfoSchemaCollector_ListSpannerSequences(t *testing.T) {
+	mockConv := &internal.Conv{
+		SpSequences: map[string]ddl.Sequence{
+			"seq1": {Name: "seq1"},
+			"seq2": {Name: "seq2"},
+		},
+	}
+	collector := InfoSchemaCollector{
+		conv: mockConv,
+	}
+	sequences := collector.ListSpannerSequences()
+	assert.Len(t, sequences, 2)
+	if _, ok := sequences["seq1"]; !ok {
+		t.Errorf("expected sequence seq1")
+	}
+	if _, ok := sequences["seq2"]; !ok {
+		t.Errorf("expected sequence seq2")
+	}
+}
+
+func TestInfoSchemaCollector_ListColumnDefinitions(t *testing.T) {
+	mockConv := &internal.Conv{
+		SrcSchema: map[string]schema.Table{
+			"table1": {
+				Id:   "table1",
+				Name: "table1",
+				PrimaryKeys: []schema.Key{
+					{ColId: "col1", Order: 1},
+				},
+				ForeignKeys: []schema.ForeignKey{
+					{Name: "fk1", ColIds: []string{"col2"}},
+				},
+				ColDefs: map[string]schema.Column{
```

Comment:
Please also add tests for failure scenarios - stuff like invalid types, no columns etc.